src/PowerShellEditorServices/Language/TokenOperations.cs

```diff
@@ -146,19 +140,19 @@ internal static class TokenOperations
         }
 
         /// <summary>
-        /// Given an array of tokens, find matching regions which start and end with a different TokenKind
+        /// Given an array of tokens, find matching regions which start (array of tokens) and end with a different TokenKind
         /// </summary>
         static private List<FoldingReference> MatchTokenElements(
             Token[] tokens,
-            TokenKind startTokenKind,
+            TokenKind[] startTokenKind,
```

Comment:
I thought about this using a HastSet instead since it's faster to look up than iterating through an array but we're talking about 3 elements so I'm not too concerned.